### ISSUE TITLE
Questions Regarding Metric Calculation in Project Replication

### ISSUE THREAD

**Issue by roal588** [ivy9092111111/RSTAR#2]
This is my first time undertaking a research project, and I am writing to consult you about some questions regarding metric calculation during the project replication process.
 
So far, I have successfully got the source code running with the sample data you provided. However, I still have some doubts about the calculation of relevant metrics, as listed below:
 
1. Is the SSIM obtained from running the code on your sample data SSIM-Global?
2. If it is, how should the remaining two metrics, namely SSIM-Tumor and SSIM-Lung, be calculated?
3. Could you please tell me the specific method you adopted to calculate RMSE?

**Comment by ZihengD** [comment]
Hi! Thanks for your questions.

1. The SSIM you see in the training script is mainly used for model training/monitoring. For the results reported in the paper (i.e., final metric evaluation), I computed SSIM offline using MATLAB. 
2. For region-specific SSIM (SSIM-Tumor and SSIM-Lung), you need binary masks for the corresponding regions and then compute SSIM within the masked region. To get the lung-region masks is straightforward, and there are many tools you may find. Tumor mask often requires manual/semi-manual segmentation. Fortunately, if you are using the same dataset as I did (the 4D-Lung dataset), it includes RTSTRUCT files that provide the tumor contours you need. In my workflow, the masks are generally defined on the ground-truth images, and then applied consistently for evaluating the reconstructed images.
3. For RMSE, I first denormalized the images so that intensities are in Hounsfield Units (HU), and then computed RMSE between the reconstructed image and the ground truth. Both MATLAB and Python have built-in implementations.

**Comment by roal588** [comment]
ok,thanks
________________________________
发件人: Ziheng Deng ***@***.***>
发送时间: 2026年1月20日 16:06
收件人: ZihengD/RSTAR ***@***.***>
抄送: roal588 ***@***.***>; Author ***@***.***>
主题: Re: [ZihengD/RSTAR] Questions Regarding Metric Calculation in Project Replication (Issue #2)

[https://avatars.githubusercontent.com/u/41174045?s=20&v=4]ZihengD left a comment (ZihengD/RSTAR#2)<https://github.com/ZihengD/RSTAR/issues/2#issuecomment-3771558208>

Hi! Thanks for your questions.

  1.  The SSIM you see in the training script is mainly used for model training/monitoring. For the results reported in the paper (i.e., final metric evaluation), I computed SSIM offline using MATLAB.
  2.  For region-specific SSIM (SSIM-Tumor and SSIM-Lung), you need binary masks for the corresponding regions and then compute SSIM within the masked region. To get the lung-region masks is straightforward, and there are many tools you may find. Tumor mask often requires manual/semi-manual segmentation. Fortunately, if you are using the same dataset as I did (the 4D-Lung dataset), it includes RTSTRUCT files that provide the tumor contours you need. In my workflow, the masks are generally defined on the ground-truth images, and then applied consistently for evaluating the reconstructed images.
  3.  For RMSE, I first denormalized the images so that intensities are in Hounsfield Units (HU), and then computed RMSE between the reconstructed image and the ground truth. Both MATLAB and Python have built-in implementations.

―

**Comment by roal588** [comment]
Hello, I'm unable to access the link to the sample dataset you posted on GitHub. Could you please send me the simulated test dataset or tell me how to obtain it?
